src/st/strategy/shortcode/class-wpml-pb-update-shortcodes-in-content.php

```diff
@@ -82,25 +82,35 @@ private function update_shortcode_attributes( $shortcode_data ) {
 
 	private function replace_string_with_translation( $block, $original, $translation, $is_attribute = false, $attr = '' ) {
 		$translation = apply_filters( 'wpml_pb_before_replace_string_with_translation', $translation, $is_attribute );
+		$has_wrapper = false;
 		$new_block   = $block;
 
 		if ( $translation ) {
 
 			if ( $this->is_string_too_long_for_regex( $original ) ) {
+				$block             = $this->remove_wrappers( $block );
 				$new_block         = str_replace( $original, $translation, $block );
 				$this->new_content = str_replace( $block, $new_block, $this->new_content );
 			} else {
 				if ( $is_attribute && $attr ) {
 					$pattern     = '/' . $attr . '=(["\'])' . preg_quote( $original, '/' ) . '(["\'])/';
 					$replacement = $attr . '=${1}' . $this->escape_backward_reference_on_replacement_string( $translation ) . '${2}';
 				} else {
+					$has_wrapper = false !== strpos( $new_block, '[' . WPML_PB_Shortcode_Content_Wrapper::WRAPPER_SHORTCODE_NAME . ']' );
 					$pattern     = '/(]\s*)' . preg_quote( trim( $original ), '/' ) . '(\s*\[)/';
 					$replacement = '${1}' . $this->escape_backward_reference_on_replacement_string( trim( $translation ) ) . '${2}';
 				}
 
-				$new_block         = preg_replace( $pattern, $replacement, $block );
-				$replacement       = $this->escape_backward_reference_on_replacement_string( $new_block );
-				$this->new_content = preg_replace( '/' . preg_quote( $block, '/' ) . '/', $replacement, $this->new_content, 1 );
+				$new_block   = preg_replace( $pattern, $replacement, $block );
+				$block       = $this->remove_wrappers( $block );
+				$new_block   = $this->remove_wrappers( $new_block );
+				$replacement = $this->escape_backward_reference_on_replacement_string( $new_block );
+
+				if ( $has_wrapper ) {
+					$this->new_content = preg_replace( '/(]\s*)' . preg_quote( $block, '/' ) . '(\s*\[)/', '${1}' . $replacement . '${2}', $this->new_content, 1 );
```

Comment:
@strategio Why is this special case here? If we have already removed the wrappers in $block and $new_block I don't understand why this special case.

Reply:
@brucepearson, since we don't have any delimiters for the string, this will ensure that the string we are replacing is (at least) inside a shortcode content (between `]` and `[`).

Otherwise, we could be replacing an attribute name (e.g. the string "title" could replace the attribute name `title`).

I will try to reflect this need in the tests.

Reply:
Added a new test @brucepearson (see https://github.com/OnTheGoSystems/wpml-page-builders/pull/48/files#diff-81155f1140cd32496c3943c384191c0bR340). This test is failing if we go through the usual case.

Reply:
I'm still struggling to easily understand the regex here. Can you move it to a private function with a meaning full name.
I still do understand don't need this special handling when we don't have wrappers. Don't we have the same problem?

Reply:
@brucepearson, when we translate a usual shortcode content, we first translate it inside its block:

`[shortcode_A]My text[/shortcode_A]` will become `[shortcode_A]FR My text[/shortcode_A]`.

Then, we replace the whole block inside the whole content.

With wrapped content, we can't do this because the wrappers do not exist in the actual content (we never inject the wrappers in the actual content). So we replace a simple string (with no delimiters) inside the whole content.

To limit possible wrong replacements (explained in https://github.com/OnTheGoSystems/wpml-page-builders/pull/48#discussion_r236255777), we make sure that this replacement is done inside a shortcode content.

I moved the replacement line into a private method `replace_content_without_delimiters`.